src/platform-implementation-js/namespaces/app-menu.ts

```diff
@@ -102,11 +104,84 @@ Items with CollapsiblePanels can also have accessories which provide primary act
 */
 export default class AppMenu {
   #driver;
+  #events = stream<
+    {
+      type: 'collapseToggled';
+      /** Whether or not the AppMenuBurger is open or not. */
+      open: boolean;
+      /**
+       * 'start' is when a `transactionstart` event is fired on a collapsible panel.
+       * 'end' is 200ms after 'start' and based on the transition time of width animation in Gmail. */
+      stage: 'start' | 'end';
+    },
+    unknown
+  >((emitter) => {
+    const f = async () => {
+      const isShown = await this.isShown();
+
+      if (!isShown) {
+        return;
+      }
+
+      const appMenu = await GmailElementGetter.getAppMenuAsync();
+
+      if (!appMenu) {
+        return;
+      }
+
+      appMenu.parentElement!.addEventListener(
+        'transitionstart',
+        async ({ target }) => {
+          if (
+            target instanceof HTMLElement &&
+            // Is the target a collapsible panel with the transition class?
+            target.matches(CollapsiblePanelView.elementSelectors.NATIVE) &&
+            target.classList.contains(
+              CollapsiblePanelView.elementCss.TOGGLE_OPEN_STATE
+            )
+          ) {
+            emitter.emit({
+              type: 'collapseToggled',
+              open: this.isMenuCollapsed(),
+              stage: 'start',
+            });
+
+            // `transitionend` TransitionEvents don't seem to fire reliably for collapsible panels. This may
```

Comment:
I believe the `transitioncancel` will be called in this case

Reply:
That does indeed seem to do the trick. See 16a0293a.